x-pack/plugins/security_solution/public/detections/components/take_action_dropdown/index.tsx

```diff
@@ -166,6 +174,19 @@ export const TakeActionDropdown = React.memo(
       onInvestigateInTimelineAlertClick: closePopoverHandler,
     });
 
+    const osqueryAvailable = osquery?.isOsqueryAvailable({
+      agentId,
+    });
+
+    const handleOnOsqueryClick = useCallback(() => {
+      onOsqueryClick(agentId);
+      setIsPopoverOpen(false);
+    }, [onOsqueryClick, agentId]);
+
+    const osqueryActionItem = OsqueryActionItem({
```

Comment:
But you might want to put the object reference here in a `useMemo` since a new props object will be created on every re-render, but it might not actually make a difference considering the spread pattern for jsx components `{ ...props }`
```
cont actionItemProps = useMemo(() => ({ handleClick: handleOnOsqueryClick }), [handleClick]); 
```

Reply:
Do you mean like this? 
```
    const osqueryActionItem = useMemo(
      () =>
        OsqueryActionItem({
          handleClick: handleOnOsqueryClick,
        }),
      [handleOnOsqueryClick]
    );
```